src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs

```diff
@@ -187,7 +187,7 @@ public static void GetVectorAndItemType(Type rawType, string name, out bool isVe
 
             if (itemType == typeof(string))
                 itemType = typeof(ReadOnlyMemory<char>);
-            else if (!itemType.TryGetDataKind(out _))
+            else if (!itemType.TryGetDataKind(out _) && TypeManager.GetDataViewTypeOrNull(itemType) == null)
```

Comment:
>else if (!itemType.TryGetDataKind(out _) && TypeManager.GetDataViewTypeOrNull(itemType) == null) [](start = 12, length = 96)

what would happen if
!itemType.TryGetDataKind(out _) && TypeManager.GetDataViewTypeOrNull(itemType) != null

Reply:
It will just not throw any error. Let me rewrite the `else if` branch here, which is only for checking if the passed-in type is known.

---
In reply to: [285271321](https://github.com/dotnet/machinelearning/pull/3263#discussion_r285271321) [](ancestors = 285271321)